craft_providers/lxd/lxc.py

```diff
@@ -1322,3 +1348,97 @@ def enable_pro_service(
                     brief=f"Failed to enable Pro service {service!r} on instance {instance_name!r}.",
                     details=errors.details_from_called_process_error(error),
                 ) from error
+
+    def is_pro_installed(
+        self,
+        *,
+        instance_name: str,
+        project: str = "default",
+        remote: str = "local",
+    ) -> bool:
+        """Check whether Ubuntu Pro Client is installed in the instance.
+
+        :param instance_name: Name of instance.
+        :param project: Name of LXD project.
+        :param remote: Name of LXD remote.
+
+        :raises LXDError: on unexpected error.
+        """
+        command = [
+            "exec",
+            f"{remote}:{instance_name}",
+            "pro",
+            "version",
+        ]
+        try:
+            self._run_lxc(
+                command,
+                capture_output=True,
+                project=project,
+            )
+
+            logger.debug("Ubuntu Pro Client is installed in managed instance.")
+            return True  # noqa: TRY300
```

Comment:
why add the `noqa` instead of doing the `else: return True`?

Reply:
because there is no `if` clause in that function.

Reply:
Oh I thought that code had an `if` before which I removed in a previous commit, but ruff couldn't forget about it :)